src/response/cache_sorted_set_fetch_response.rs

```diff
@@ -1,7 +1,159 @@
-use momento_protos::cache_client::SortedSetElement;
+use crate::sorted_set::Elements;
+use crate::MomentoError;
+use core::convert::TryFrom;
+use momento_protos::cache_client::sorted_set_fetch_response::SortedSet;
 
 #[derive(Debug)]
 #[non_exhaustive]
 pub struct MomentoSortedSetFetchResponse {
-    pub value: Option<Vec<SortedSetElement>>,
+    pub value: Option<SortedSet>,
+}
+
+pub enum SortedSetFetch {
+    Found { elements: Elements },
+    Missing,
+}
+
+impl TryFrom<SortedSetFetch> for Vec<(Vec<u8>, f64)> {
+    type Error = MomentoError;
+
+    fn try_from(value: SortedSetFetch) -> Result<Self, Self::Error> {
+        match value {
+            SortedSetFetch::Found { elements } => match elements {
+                Elements::ValuesWithScores(mut values) => {
+                    let result: Vec<(Vec<u8>, f64)> = values
+                        .elements
+                        .drain(..)
+                        .map(|e| (e.value, e.score))
+                        .collect();
+                    Ok(result)
+                }
+                Elements::Values(_) => Err(MomentoError::TypeError {
+                    description: std::borrow::Cow::Borrowed(
+                        "response did not contain element scores",
+                    ),
+                    source: Box::new(std::io::Error::new(
+                        std::io::ErrorKind::InvalidData,
+                        "element did not contain a score",
+                    )),
+                }),
+            },
+            SortedSetFetch::Missing => Err(MomentoError::Miss {
+                description: std::borrow::Cow::Borrowed("sorted set was not found"),
+            }),
+        }
+    }
+}
+
+impl TryFrom<SortedSetFetch> for Vec<Vec<u8>> {
```

Comment:
If I'm reading this right, this would correspond to `with_scores=false` on the request, which we don't support for now, so you can probably get rid of this?